File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        int n = nums.length;
        for (int i = 0;i <= n; i++) {
            int dif = target - nums[i]; 
            if (seen.containsKey(nums[i])) return new int[] {seen.get(nums[i]), i};
            else seen.put(dif, i);
        }

        return new int[] {};
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-2.kt
class Solution {
    fun search(nums: IntArray, target: Int): Int {
        var l = 0
        var r = nums.size - 1
        while (l <= r) {
            val mid = (l+r)/2
            if (nums[mid] == target) return mid
            if (nums[mid] > nums[r]) {
                if (nums[l] <= target && nums[mid] > target) {
                    r = mid - 1
                } else {
                    l = mid + 1
                }
            } else {
                if (nums[mid] < target && nums[r] >= target) {
                    l = mid + 1
                } else {
                    r = mid - 1
                }
            }
        }

        return -1
    }
}
